src/main/scala/projects.scala

```diff
@@ -75,7 +75,9 @@ trait AnyTask extends AnyType {
   type Input <: AnyDataSet
   val input: Input
 
-  type Output <: AnyDataSet
+  // FIXME: this is supposed to be the same:
+  // type Output <: AnyDataSet // = AnyRecordType.Of[AnyData]
+  type Output <: AnyRecordType { type Keys <: AnyProductType { type Types <: AnyKList { type Bound <: AnyData } } }
```

Comment:
I don't understand why it doesn't compile with `AnyDataSet`, which is `AnyRecordType.Of[AnyData]`, which is supposed to be the same as the last line (which makes it compile):

```scala
case object AnyRecordType {
  type Of[+B <: AnyType] = AnyRecordType { type Keys <: AnyProductType.Of[B] }
}
case object AnyProductType {
  type Of[+B <: AnyType] = AnyProductType { type Types <: AnyKList.Of[B] }
}
case object AnyKList {
  type Of[+B] = AnyKList { type Bound <: B }
}
```

@eparejatobes any clue?

Reply:
There are a lot of issues with inference and aliasing. A sample:

- https://github.com/scala/bug/issues/8709
- https://github.com/scala/bug/issues/8740
- https://github.com/scala/bug/issues/10238
- https://github.com/scala/bug/issues/10528
- https://github.com/scala/bug/issues/10556
- ...